sinonjs/sinon
# stub.throws and TypeErrors

Docs and tests misleadingly suggest that string parameters will generate
an error of that type. Documentation and test changes should make
clear(er) that string params supply generic error name and/or message
parameters.

#### Purpose (TL;DR) - mandatory
Docs and tests misleadingly suggest that string parameters will generate an error of that type

Example from stubs.md:
```
stub.throws("TypeError");

Causes the stub to throw an exception of the provided type.
```

Stub does not throw a "TypeError" exception. Instead, it throws an Error with the `name` property set to "TypeError"



#### Solution  - optional

Relevant docs and unit-tests have had the magic string "TypeError" changed to "Named Error".


#### How to verify - mandatory
1. Check out this branch
2. `npm install`
3. `npm test`

NOTE: existing issue related to `referee` requires workaround to successfully run all tests. This workaround (updating afftected file to use `@sinonjs/referree`) is not part of the PR.

#### Checklist for author
